src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,179 +1,280 @@
+    private val pricePattern by repoGroup.pattern("price", "([0-9]{2,3}K|[0-9]{1,3}M|[0-9]+\\.[0-9]M|[0-9] ?MIL)")
+    private val carryPattern by repoGroup.pattern("carry", "(CARRY|CARY|CARRIES|CARIES|COMP|TO CATA [0-9]{2})")
+    private val nonPugPattern by repoGroup.pattern("nonpug", "(PERM|VC|DISCORD)")
+    private val memberPattern by repoGroup.pattern("member", ".*§.(?<playerName>.*)§f: §e(?<className>.*)§b \\(§e(?<level>.*)§b\\)")
+    private val ineligiblePattern by repoGroup.pattern(
+        "ineligible",
+        "^§c(Requires .*$|You don't meet the requirement!|Complete previous floor first!$)"
+    )
+    private val classLevelPattern by repoGroup.pattern(
+        "classlevel",
+        " §.(?<playerName>.*)§f: §e(?<className>.*)§b \\(§e(?<level>.*)§b\\)"
+    )
+    private val notePattern by repoGroup.pattern("note", "^(§7§7Note: |§f[^§])")
+    private val floorTypePattern by repoGroup.pattern("floortype", "(The Catacombs).*|.*(MM Catacombs).*")
+    private val checkIfPartyPattern by repoGroup.pattern("checkifparty", ".*('s Party)")
+    private val partyFinderTitlePattern by repoGroup.pattern("pfindertitle", "(Party Finder)")
+    private val catacombsGatePattern by repoGroup.pattern("catagate", "(Catacombs Gate)")
+    private val selectFloorPattern by repoGroup.pattern("selectfloor", "(Select Floor)")
+    private val entranceFloorPattern by repoGroup.pattern("entrance", "(.*Entrance)")
+    private val floorPattern by repoGroup.pattern("floor", "(Floor .*)")
+    private val anyFloorPattern by repoGroup.pattern("anyfloor", "(Any)")
+    private val masterModeFloorPattern by repoGroup.pattern("mmfloor", "(MM )|(.*Master Mode Catacombs)")
+    private val dungeonFloorPattern by repoGroup.pattern("dungeonfloor", "(Dungeon: .*)")
+    private val floorFloorPattern by repoGroup.pattern("floorpattern", "(Floor: .*)")
 
+    //  Variables used
     private var selectedClass = ""
+    private var floorStackSize = mutableMapOf<Int, String>()
+    private var highlightParty = mutableMapOf<Int, LorenzColor>()
+    private var toolTipMap = mutableMapOf<Int, MutableList<String>>()
+    private var inInventory = false
 
     @SubscribeEvent
-    fun onRenderItemTip(event: RenderItemTipEvent) {
-        if (!LorenzUtils.inSkyBlock || LorenzUtils.skyBlockArea != "Dungeon Hub") return
-        if (!config.floorAsStackSize) return
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+        val inventoryName = event.inventoryName
 
-        val itemName = event.stack.name.removeColor()
-        val invName = InventoryUtils.openInventoryName()
+        DelayedRun.runDelayed(10.0.milliseconds) {
+            floorStackSize.clear()
+            highlightParty.clear()
+//
+//        Code for Stack Tip
+//
+            if (catacombsGatePattern.matches(inventoryName)) {
+                val lore = event.inventoryItems[45]?.getLore()
```

Comment:
I don't like magic numbers. Please make a constant for the 45 with a meaningful name.